### PR TITLE
[#179810521] Cleanup old queries

### PR DESCRIPTION
**Pull request by prajjwolmondal**
[Ticket](https://www.pivotaltracker.com/story/show/179810521)

# Description

Implemented a `@Scheduled` method that kills queries that haven't been accessed in the last 2 mins. This is done by storing the `nextPageUrl` for all queries in the `query_job` table, and then sending a `DELETE` request to that URL to kill the query.

Full change-log:
- Updated the `query_job` table to:
    - record timestamps for when a query was started, finished, and last accessed. 
    - record the `next_page_url` for the query 
- The `id` stored in the `query_job` was randomly generated but I've modified it to use the query_id returned by Trino. This was done to support my initial approach (send a kill query command with the query_id) but I still feel like it should be query_id rather than a randomly generated ID. Let me know if this should be reverted.
- The `@Scheduled` method is set to run every 15 mins, and the default timeout for queries to kill is set to 2 mins. Both these are configurable via `application.yml`

### CODE REVIEW

File: src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoDataConnectAdapter.java [hunk 1]
```diff
@@ -285,9 +286,8 @@ public TableData search(
                 "query", Optional.ofNullable(rewrittenQuery).orElse("(undefined)"))
         );
         JsonNode response = client.query(rewrittenQuery, extraCredentials);
-        QueryJob queryJob = createQueryJob(query, dataModel);
-        TableData tableData = toTableData(NEXT_PAGE_SEARCH_TEMPLATE, response, queryJob.getId(), request);
-        return tableData;
+        QueryJob queryJob = createQueryJob(response.get("id").asText(), query, response.get("nextUri").asText(), dataModel);
```
Comment (sharvari-kapadia) [inline src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoDataConnectAdapter.java:289]: @jvlasblom Any objection on using trino query id vs generating one ourself? Wanted to get your opinion as you are the original author of `query_job` table.

File: src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoHttpClient.java [hunk 2]
```diff
@@ -251,9 +263,9 @@ private Response execute(final Request.Builder request, Map<String, String> extr
                 .size());
             Response response = httpClient.newCall(r).execute();
             log.info("GET "+r.url()+" returned "+response.code());
-            if(response !=null && !response.isSuccessful()){
+            if(response != null && !response.isSuccessful()){
```
Comment (sharvari-kapadia) [inline src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoHttpClient.java:266]: :)

**Review by sharvari-kapadia — APPROVED** [review]
Some nit comments.
Overall LGTM.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -40,6 +40,8 @@ app:
     oauth-client:
       resource: "${app.indexing-service.base-uri}"
     publisher-catalog-name: publisher_data
+  query-cleanup-interval: "0 */15 * ? * *" # Every 15 mins
```
[the remark this answers is not in the document]
Comment (prajjwolmondal) [reply src/main/resources/application.yml:43]: As per [Jonathans message](https://www.pivotaltracker.com/story/show/179810521/comments/228945651) I've bumped it down to run every 10 secs

File: src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoDataConnectAdapter.java (hunk 1 above)
Comment (jvlasblom) [reply src/main/java/com/dnastack/ga4gh/dataconnect/adapter/trino/TrinoDataConnectAdapter.java]: No objection

**Pull request merged**